kernel/kernel-api/src/main/java/io/delta/kernel/internal/tablefeatures/TableFeatures.java

```diff
@@ -235,7 +235,8 @@ private static class VariantShreddingPreviewFeature extends TableFeature.ReaderW
 
     @Override
     public boolean metadataRequiresFeatureToBeEnabled(Protocol protocol, Metadata metadata) {
-      return TableConfig.VARIANT_SHREDDING_ENABLED.fromMetadata(metadata);
+      return hasTypeColumn(metadata.getSchema(), VARIANT)
```

Comment:
what happens if the variantType table feature is supported, but there is no variant column in it?

Do we expect to not enable the shredding feature even when the user explicitly sets the table property?

Reply:
yeah, the shredding table feature won't be enabled in this case, we'll only add the shredding table feature if there is actually variants in the table. This helps ensure that older readers that may not recognize the `variantShredding` table feature can still read tables if the shredding property is for some reason enabled, but there aren't any variants in the schema